src/celeritas/em/model/GammaNuclearModel.cc

```diff
@@ -98,5 +111,49 @@ void GammaNuclearModel::step(CoreParams const&, CoreStateDevice&) const
 }
 #endif
 
+//---------------------------------------------------------------------------//
+/*!
+ * Build CHIPS gamma-nuclear element cross sections using G4GammaNuclearXS.
+ */
+inp::Grid
+GammaNuclearModel::calc_chips_xs(AtomicNumber z, double emin, double emax) const
+{
+    CELER_EXPECT(z);
+
+    inp::Grid result;
+
+    // Tabulate cross sections using separate parameterizations for the high
+    // energy region (emin < E < 50 GeV) and the ultra high energy region up
+    // to the maximum valid energy (emax). The numbers of bins are chosen to
+    // adequately capture both the parameterized points (224 bins from 106 MeV
+    // to 50 GeV) and the calculations used in G4PhotoNuclearCrossSection,
+    // which can be made configurable if needed (TODO). Note that the linear
+    // interpolation between the upper limit of the IAEA cross-section data
+    // and 150 MeV, as used in G4GammaNuclearXS, is also included in this
+    // tabulation.
+    double const emid = 5e+4;
+    size_type nbin_total = 300;
+    size_type nbin_ultra = 50;
+
+    result.x.resize(nbin_total);
+    result.y.resize(nbin_total);
+
+    result.x = geomspace(emin, emid, nbin_total - nbin_ultra);
+    result.x.pop_back();
+    auto ultra = geomspace(emid, emax, nbin_ultra + 1);
+    result.x.insert(result.x.end(), ultra.begin(), ultra.end());
+
+    // Tabulate the cross section from emin to emax
+    EmExtraPhysicsHelper::MmSqXs xs;
+    for (size_type i = 0; i < nbin_total; ++i)
+    {
+        xs = helper_->calc_gamma_nuclear_xs(z, MevEnergy{result.x[i]});
```

Comment:
Since it's only used inside the loop, to simplify scope:
```suggestion 
    for (size_type i = 0; i < nbin_total; ++i)
    {
        auto xs = helper_->calc_gamma_nuclear_xs(z, MevEnergy{result.x[i]});
```

Reply:
SImplified the code as suggested.